static_report/src/utils/index.tsx

```diff
@@ -138,59 +145,77 @@ export function extractExpectedOrActual(value) {
 }
 
 // for comparison
-export function joinBykey(base = {}, input = {}) {
-  const result = {};
-
-  Object.entries(base).forEach(([key, value]) => {
-    if (!result[key]) {
-      result[key] = {};
+export function nestComparisonValueByKey(
+  base: Record<string, unknown>,
+  input: Record<string, unknown>,
+): Record<
+  string,
+  { base: Record<string, unknown>; input: Record<string, unknown> }
+> {
+  const _base = Object.entries(base).reduce((acc, [key, value]) => {
+    if (!acc[key]) {
+      acc[key] = Object.create({});
     }
 
-    result[key]['base'] = value;
-  });
+    acc[key] = value;
```

Comment:
hmm, did we lose the `['base' | 'input']` key access here?

Reply:
perhaps write a function for `.reduce(myFunc('input'), {})`
```
const myFunc = (type) => (acc, [key, val]) => { 
  acc[key] = acc[key][type] ? Object.create({}) : value } 
  return acc;
}
```

Reply:
In the `mergeWith`:

```js
{
  base: base,
  input: input
}
```